File: Java/Problems/Arrays/KSubstringVowels.java
package Problems.Arrays;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/**
 * k-Substring Vowels
 * Medium
 * 0 / 50
 * Given a string s and a number k, find the number of vowels in every substring of size k.
 *
 * Vowels: ['a', 'e', 'i', 'o', 'u']
 */
public class KSubstringVowels {

    int[] kSubstringVowels (String s, int k) {
        // add your logic here
        List<Integer> list = new ArrayList<>();
        char[] c = s.toCharArray();
        int count;
        for (int i=0; i<=c.length-k; i++){
            count=0;
            for (int j=i; j<i+k; j++){
//                System.out.println("->before"+c[j]);
                switch (c[j]){
                    case 'a':
                    case 'e':
                    case 'i':
                    case 'o':
                    case 'u': {
                        count++;
                        break;
                    }
                }
            }
            list.add(count);
        }
        return list.stream().mapToInt(i->i).toArray();
    }

    public static void main(String[] args) {
        KSubstringVowels kSubstringVowels = new KSubstringVowels();

        Arrays.stream(kSubstringVowels.kSubstringVowels("workattech", 3))
                .forEach(i -> System.out.println(i));

    }
}

File: GO-Lang/Problems/Arrays/KSubstringVowels/KSubstringVowels.go
package main

import "fmt"

func KSubstringVowels(s string, k int) []int {
	count := 0
	counts := []int{}
	chars := []rune(s)

	for i := 0; i < len(chars)-k; i++ {
		count = 0
		for j := i; j < i+k; j++ {
			if string(chars[j]) == "a" || string(chars[j]) == "e" || string(chars[j]) == "i" || string(chars[j]) == "o" || string(chars[j]) == "u" {
				count++
			}
		}
		counts = append(counts, count)
	}
	return counts
}

func main() {
	fmt.Println("--->", KSubstringVowels("workattech", 3))
}
